src/main/java/org/gelecekbilimde/scienceplatform/auth/controller/AuthController.java

```diff
@@ -67,4 +71,34 @@ public Response<Void> logout(@RequestBody @Valid LogoutRequest logoutRequest) {
 		return Response.NO_CONTENT;
 	}
 
+	@PutMapping("/requestToBeAuthor")
+	@PreAuthorize("hasRole('USER')")
+	public Response<UserRoleResponse> requestToBeAuthor() {
+		return Response.ok(roleService.userRoletoAuthorRoleRequest());
+	}
+
+	@PutMapping("/requestToBeAdmin")
+	@PreAuthorize("hasRole('ADMIN')")
+	public Response<UserRoleResponse> requestToBeAdmin(@RequestBody @Valid RoleChangeRequest request) {
+		return Response.ok(roleService.makeUserToAdmin(request));
+	}
+
+	@PutMapping("/makeUserAuthor")
+	@PreAuthorize("hasRole('ADMIN')")
+	public Response<UserRoleResponse> makeUserAuthor(@RequestBody @Valid RoleChangeRequest request) {
+		return Response.ok(roleService.makeUserToAuthor(request));
+	}
+
+	@GetMapping("/getRequestToBeAuthor")
+	@PreAuthorize("hasRole('ADMIN')")
+	public Response<List<UserRoleResponse>> getAllRequestToBeAuthor() {
+		return Response.ok(roleService.getAllUserRoletoAuthorRoleRequest());
+	}
+
+	@DeleteMapping("/deleteRequestToBeAuthor")
+	@PreAuthorize("hasRole('ADMIN')")
+	public Response deleteRequestToBeAuthor(@RequestBody @Valid RoleChangeRequest request) {
+		return Response.ok(roleService.deleteUserRoletoAuthorRoleRequest(request));
+	}
```

Comment:
Rol için ayrı bir controller oluşturalım, oraya taşıyalım endpointleri, sonrasında endpointleri kararlaştırmak üzerine bi konuşalım çünkü rest standartlarına uymuyor, düzenlemek gerekiyor mantığına göre

Reply:
hangi package altında olmalı role controller?

Reply:
`auth.controller` altında olabilir. Bütün auth işlemleri burada yapılıyor.

Reply:
RoleController adında ayrı bir controller olmasına katılıyorum. MakeRequestToBeAdmin'den ziyade MakeRequestToBeModerator olabilir. Moderator rolü eklenebilir. Issue'nın discussion'ınına yazmıştım ancak tekrar yazayım. Ayrıca makeUserAuthor ve makeUserAdmin yerine tek bir changeUserRole methodu yazılabilir ve bu 
method tüm rol geçişleri için kullanılabilir. Bugün isterler için dökümanı oluşturacağım onu da eklerim daha sonrasında.
Ana olarak 5 tane rolümüz olmalı.
User: Post okuma, takip etme, beğenme, yorum yapma ve kendi profilini editleme yetkisi var.
Author: User + yazı yazma, kendi yazısını editleme
Moderator: Author + tüm yazıları editleme
Admin : tüm permissionlara sahip

Reply:
Bu rolleri oluşturma işi issuede yoktu farklı bir issue'da vardı ondan dolayı oluşturmadım iki issueyi de yapmam gerekiyorsa yapayım.

Reply:
Bir de changeUserRole oluşturursak hasAuhtorty'yi kime açacağız sadece admine mi? Çünkü bu methodu kullanırsak bütün rol geçişlerini yapabilecek bir rolde olmalı.

Reply:
Senin görevinin altına yorum olarak yazmıştım. Asıl istek admin olmak için değil moderatör olmak için olabilir. Admin olarak admin kişi kendisi rolü değiştirirken kişiyi isterse admin yapabilir. Moderatör olmak isteme requestini de sadece Yazarlar yapmalıdır.

Reply:
> Bir de changeUserRole oluşturursak hasAuhtorty'yi kime açacağız sadece admine mi? Çünkü bu methodu kullanırsak bütün rol geçişlerini yapabilecek bir rolde olmalı.

Aynen admin yapabilir.

Reply:
Tamam o zaman o şekilde düzenlerim.

Reply:
@meminkecik farklı bir controller oluşturmuşuz ama buradaki endpointler silinmemiş